cranelift/filetests/filetests/isa/x64/i128.clif

```diff
@@ -440,15 +440,15 @@ block0(v0: i128, v1: i128):
 ;   cmpq %rdx, %rdi
 ;   sbbq %rcx, %rsi
 ;   setae %dil
-;   andl %r9d, %eax
-;   andl %r11d, %r8d
-;   andl %r14d, %r10d
-;   andl %ebx, %r13d
-;   andl %edi, %r15d
-;   andl %r8d, %eax
-;   andl %r13d, %r10d
-;   andl %r10d, %eax
-;   andl %r15d, %eax
+;   andb %r9b, %al
+;   andb %r11b, %r8b
+;   andb %r14b, %r10b
+;   andb %bl, %r13b
+;   andb %dil, %r15b
+;   andb %r8b, %al
+;   andb %r13b, %r10b
+;   andb %r10b, %al
+;   andb %r15b, %al
```

Comment:
(sorry if this was already covered elsewhere) but how come these are changing from a 32-bit and to an 8-bit and?

Reply:
I'm a bit mystified: CLIF's `icmp` should return a truthy `Int` which reduces down to an 8-bit value here:

https://github.com/bytecodealliance/wasmtime/blob/a0338af84f66cb452fdf4b692d4facb5d052c12d/cranelift/codegen/meta/src/cdsl/typevar.rs#L433-L435

So those `band` instructions should always have lowered to an `andb`? Perhaps the original encoding just modified a larger width...

Reply:
I talked a bit more about this with @abrown and the reason for this is that Cranelift today [clamps operand sizes to 32-bits](https://github.com/bytecodealliance/wasmtime/blob/a0338af84f66cb452fdf4b692d4facb5d052c12d/cranelift/codegen/src/isa/x64/inst.isle#L2061) in the `alu_rmi_r` constructor. That means that all 8-bit `and`s are actually 32-bit `and`s when emitted to x64. Andrew's PR is (IMO correctly) changing the 8-bit and to use `andb` instead. 

@cfallin do you remember what `operand_size_of_type_32_64` was doing there? For example why clamp it to 32-bit? (and if so, are said reasons still applicable today?)

Reply:
Ok, see 2c8a276.